app/Platform/Controllers/SystemController.php

```diff
@@ -76,16 +83,112 @@ public function destroy(System $system): void
+            'filter.populated' => 'sometimes|string|in:yes,no,all',
+            'filter.status' => 'sometimes|string|in:all,unstarted,lt-beaten-softcore,gte-beaten-softcore,gte-beaten-hardcore,gte-completed,eq-mastered,eq-beaten-softcore-or-beaten-hardcore,any-softcore,revised',
+        ]);
+        $sortOrder = $validatedData['sort'] ?? 'title';
+        $filterOptions = [
+            'populated' => $validatedData['filter']['populated'] ?? 'yes',
+        ];
+        if (isset($validatedData['filter']['status'])) {
+            $filterOptions['status'] = $validatedData['filter']['status'];
+        }
+
+        $gameIds = $system->games()->get(['ID'])->pluck('ID')->toArray();
+        $totalUnfilteredCount = count($gameIds);
+
+        $this->gameListService->initializeUserProgress($loggedInUser, $gameIds);
+        $this->gameListService->initializeGameList($gameIds);
+
+        if (isset($filterOptions['populated'])) {
+            $this->gameListService->filterGameList(function ($game) use ($filterOptions) {
+                return $this->usePopulatedFilter($game, $filterOptions['populated']);
+            });
+        }
+
+        if (isset($filterOptions['status'])) {
+            $this->gameListService->filterGameList(function ($game) use ($filterOptions) {
+                return $this->gameListService->useGameStatusFilter($game, $filterOptions['status']);
+            });
+        }
+
+        $this->gameListService->mergeWantToPlay($loggedInUser);
+        $this->gameListService->sortGameList($sortOrder);
+
+        $availableSorts = $this->gameListService->getAvailableSorts();
+        $availableCheckboxFilters = [];
+        $availableRadioFilters = [
+            [
+                'kind' => 'populated',
+                'label' => 'Has achievements',
+                'options' => [
+                    'yes' => 'Yes',
+                    'no' => 'No',
+                    'all' => 'Either',
+                ],
+            ],
+        ];
+
+        $availableSelectFilters = [];
+        if ($loggedInUser) {
+            $availableSelectFilters = [
+                [
+                    'kind' => 'status',
+                    'label' => 'Status',
+                    'options' => [
+                        'all' => 'All games',
+                        'unstarted' => 'No achievements earned',
+                        'lt-beaten-softcore' => 'Has progress, but no award',
+                        'gte-beaten-softcore' => 'Beaten (softcore) or greater',
+                        'gte-beaten-hardcore' => 'Beaten or greater',
+                        'eq-beaten-softcore-or-beaten-hardcore' => 'Beaten, but still missing achievements',
+                        'gte-completed' => 'Completed or mastered',
+                        'revised' => 'Completed or mastered, but the set was revised',
```

Comment:
Also likely unrelated: I have a completed game that's mostly hardcore and hasn't been revised, but it's showing up under "revised":
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/c450d1ce-4660-4eb2-8b2a-80639931ce0a)

Reply:
Yep, another logic error here. Should be fixed in latest.